include/nil/crypto3/zk/snark/systems/plonk/placeholder/preprocessor.hpp

```diff
@@ -386,20 +386,26 @@ namespace nil {
                         std::array<polynomial_dfs_type, 2> &q_last_q_blind,
                         const typename ParamsType::commitment_params_type &commitment_params) {
 
-                        std::vector<polynomial_dfs_type> fixed_polys = id_perm_polys;
-                        fixed_polys.insert( fixed_polys.end(), sigma_perm_polys.begin(), sigma_perm_polys.end() );
-                        for (std::size_t i = 0; i < public_table.constants().size(); i ++){
-                            fixed_polys.push_back(public_table.constants()[i]);
+                        std::vector<polynomial_dfs_type> fixed_polys(id_perm_polys.size()+sigma_perm_polys.size()+public_table.constants().size()+public_table.selectors().size()+2);
+                        std::size_t p = 0;
+                        for(auto &i: id_perm_polys) {

```

Comment:
Please use std::copy here, for better readability.

Reply:
updated in 9bb04400